pkg/controller/endpointslice/reconciler.go

```diff
@@ -216,6 +222,13 @@ func (r *reconciler) reconcileByAddressType(service *corev1.Service, pods []*cor
 		})
 	}
 
+	// New slices must inherit current slices annotations.
```

Comment:
-1 on this behavior. If an external tool wants to observe/annotate generated endpointslices for some reason, I guess they are free to do so, and the endpointslice controller should leave those annotations alone, but propagating annotations between peer generated endpointslices does not match my expectations.

This would be like the replicaset controller propagating custom annotations from one child pod into other pods it created for the same replicaset... that doesn't make a lot of sense to me.

Reply:
Agreed - users should be able to annotate any objects in kube, and controllers should respect that (but may stomp if they leverage that annotation for some reason), but propagation across unrelated instances should only occur when that is a defined feature of the object itself (and it’s hard to come up with an example where it would move across instances)